exonum/src/storage/proof_map_index/proof.rs

```diff
@@ -460,6 +460,20 @@ impl<K, V> MapProof<K, V> {
     pub fn missing_keys_unchecked(&self) -> Vec<&K> {
         self.entries.iter().filter_map(|e| e.as_missing()).collect()
     }
+
+    /// Retrieves references to existing and non-existing entries in the proof.
+    ///
+    /// Existing entries have `Some` value, non-existing have `None`.
+    /// This method does not perform any integrity checks of the proof.
+    pub fn all_entries_unchecked(&self) -> Vec<(&K, Option<&V>)> {
```

Comment:
I wonder if returning `impl Iterator<Item = (&K, Option<&V>)>` could cover more use cases. When other `unchecked` methods were written, `impl Trait` wasn't in stable Rust yet, so they used `Vec<_>` out of necessity. Also, it could make `missing_keys_unchecked()` method easily implementable as a filter:

```rust
pub fn missing_keys_unchecked(&self) -> impl Iterator<Item = &K> {
    self.all_entries_unchecked().filter_map(|pair| match pair {
        (key, None) => Some(key),
        _ => None,
    })
}
```

Reply:
Seems reasonable to me, but we have some issues with supporting latest Rust version in EJB. It will be fixed before we'll use Exonum 0.10 (where this PR will be landed), but right now I wouldn't be able to test the functionality. So I'll check it and maybe fix it.

Reply:
Sure. Just to clarify things, `impl Trait` [was added in Rust 1.26](https://blog.rust-lang.org/2018/05/10/Rust-1.26.html), which seems to be the version used by EJB, at least judging [by the Travis config](https://github.com/exonum/exonum-java-binding/blob/5df785dcaae87d7fa9c4ab9832b9218dc41e39d8/.travis.yml#L20).